typeform/typeform.py

```diff
@@ -74,20 +70,48 @@ def read(self, n = None):
             form['offset'] += FETCH_LIMIT
 
 
-        # we only need the 'questions' and 'responses' results
-        # that returned from the server. On top of that, add the
-        # destination table name
+        # general results statistics records
+        stats = {
+            '__table': 'stats',
+            '__form': form.get('name'),
+            'total': stats.get('total'),
+            'completed': stats.get('completed'),
+            'id': form.get('value')
+        }
+        stats = [stats]
+
+        # helper method that assists us to add the destination table,
+        # form name and generated unique id for a given item.
+        def add_attrs(dest, id_suffix):
+            def x(item):
+                item['__table'] = dest
+                item['__form'] = form.get('name')
+
+                # we're saving the untouched original id as it comes
+                # back from the server
+                if 'id' in item:
+                    item['oid'] = item.get('id')
+
+                # inject a custom id that's assembled from the form
+                # id and a given unique attribute (e.g. token)
+                item['id'] = '%s_%s' % (form.get('value'), item.get(id_suffix))
 
-        # add the questions records
-        dest = (form['name'] + '_questions').replace(' ', '_')
-        results = map(lambda x: dict(__table=dest, **x), body['questions'])
+                return item
 
-        # add the responses (answers) records
-        dest = (form['name'] + '_responses').replace(' ', '_')
-        responses = map(lambda x: dict(__table=dest, **x), body['responses'])
+            return x
+
+        # questions records
+        questions = map(add_attrs('questions', 'field_id'),
+            body.get('questions')
```

Comment:
just in case `questions` is not defined it may be better to write it like `body.get('questions', [])` so the map won't fail with error